store/sqlstore/channel_store.go

```diff
@@ -1630,6 +1630,60 @@ func (s SqlChannelStore) UpdateMember(member *model.ChannelMember) (*model.Chann
 	return updatedMembers[0], nil
 }
 
+func (s SqlChannelStore) UpdateMemberNotifyProps(channelID, userID string, props map[string]string) (*model.ChannelMember, error) {
+	tx, err := s.GetMaster().Begin()
+	if err != nil {
+		return nil, errors.Wrap(err, "begin_transaction")
+	}
+	defer finalizeTransaction(tx)
+
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		_, err = tx.Exec(`UPDATE channelmembers
+			SET notifyprops = notifyprops || $1::jsonb
+			WHERE userid=$2 AND channelid=$3`, model.MapToJson(props), userID, channelID)
+	} else {
+		// It's difficult to construct a SQL query for MySQL
+		// to handle a case of empty map. So we just ignore it.
+		if len(props) > 0 {
+			// unpack the keys and values to pass to MySQL.
+			args := make([]interface{}, 0, len(props))
+			for k, v := range props {
+				args = append(args, "$."+k)
+				args = append(args, v)
+			}
+			args = append(args, userID, channelID)
+
+			// We calculate the number of ? to set in the query string.
+			argString := strings.Repeat("?, ", len(props)*2)
+			// Strip off the trailing comma.
+			argString = argString[:len(argString)-2]
```

Comment:
Would it be useful to turn the `args` and `argString` construction into some sort of util function that other stores can use? Seems like this will be a common pattern when dealing with a JSON field

Reply:
Good point. I think that makes sense.